splunk-otel-android/src/main/java/com/splunk/rum/SpanFilterBuilder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * Allows to modify span data before it is sent to the exported. Spans can be modified or entirely
+ * rejected from export.
+ */
+public final class SpanFilterBuilder {
+
+    private Predicate<String> rejectSpanNamesPredicate = spanName -> false;
+    private final Map<AttributeKey<String>, Predicate<String>> rejectSpanAttributesPredicates = new HashMap<>();
+    private final Map<AttributeKey<String>, Function<String, String>> spanAttributeReplacements = new HashMap<>();
+
+    SpanFilterBuilder() {
+    }
+
+    /**
+     * Remove matching spans from the exporter pipeline.
+     * <p>
+     * Span names that match the {@code spanNamePredicate} will not be exported.
+     *
+     * @param spanNamePredicate A function that returns true if a span with passed name should be
+     *                          rejected.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder rejectSpansByName(Predicate<String> spanNamePredicate) {
+        rejectSpanNamesPredicate = rejectSpanNamesPredicate.or(spanNamePredicate);
+        return this;
+    }
+
+    /**
+     * Remove matching spans from the exporter pipeline.
+     * <p>
+     * Span that contains an attribute with key {@code attributeKey} and value matching the
+     * {@code attributeValuePredicate} will not be exported.
+     *
+     * @param attributeKey            An attribute key to match.
+     * @param attributeValuePredicate A function that returns true if a span containing an attribute
+     *                                with matching value should be rejected.
+     * @return {@code this}.
+     */
+    public SpanFilterBuilder rejectSpansByAttributeValue(
+            AttributeKey<String> attributeKey,
+            Predicate<String> attributeValuePredicate) {
+
+        rejectSpanAttributesPredicates.compute(attributeKey,
+                (k, oldValue) -> oldValue == null
+                        ? attributeValuePredicate
+                        : oldValue.or(attributeValuePredicate));
+        return this;
+    }
+
+    public SpanFilterBuilder removeSpanAttribute(AttributeKey<String> attributeKey) {
```

Comment:
this is missing javadoc.   But more importantly, are we sure that `<String>` is good enough here? Why not any type?

Reply:
Added javadoc